### PR TITLE
language: add daml-react package to ts libraries

### PR DESCRIPTION
**Pull request by robin-da**
This adds the library formerly known as `daml-react-hooks` into the
monorepo. We renamed it to `@daml/react`.

The tests sadly don't work with bazel right now because the local
imports aren't resolved correctly. Local testing with `yarn run test`
works as usual. We're working on the bazel test rules in the meantime.
 
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: language-support/ts/daml-react/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    tsconfig = ":tsconfig.json",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//language-support/ts/daml-ledger",
+        "//language-support/ts/daml-types",
+        "@language_support_ts_deps//:node_modules",
+    ],
+) if not is_windows else None
+
+genrule(
```
Comment (cocreature) [inline language-support/ts/daml-react/BUILD.bazel]: I guess if you just reference `//:LICENSE` in `data` it ends up in the wrong place? would be good to document this in a comment.

File: language-support/ts/daml-react/credentials.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { decode } from 'jwt-simple';
+
+const LEDGER_ID: string = 'DAVL';
```
Comment (cocreature) [inline language-support/ts/daml-react/credentials.ts]: This should probably go away before we publish this as a library.

File: language-support/ts/daml-react/BUILD.bazel (hunk 1 above)
Comment (robin-da) [reply language-support/ts/daml-react/BUILD.bazel]: yes, is the same in all the ts libraries :panda_face: . I'll put a comment.

File: language-support/ts/daml-react/credentials.ts (hunk 2 above)
Comment (robin-da) [reply language-support/ts/daml-react/credentials.ts]: good catch, thanks!

File: language-support/ts/daml-react/credentials.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+/**
+ * Check that the party in the token matches the party of the credentials and
+ * that the ledger ID in the token matches the given ledger id.
+ */
+export const preCheckCredentials = ({party, token, ledgerId}: Credentials): string | null => {
+  const decoded = decode(token, '', true);
+  if (!decoded.ledgerId || decoded.ledgerId !== ledgerId) {
+    return 'The password is not valid for DAVL.';
```
Comment (cocreature) [inline language-support/ts/daml-react/credentials.ts]: This also mentions DAVL. If there is more stuff that references DAVL I’m happy for this to be addressed in a separate PR. probably good to just grep through the codebase before we publish the library.

Comment (robin-da) [reply language-support/ts/daml-react/credentials.ts]: thanks, fixed.

File: language-support/ts/daml-react/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "esModuleInterop": true,
+    "outDir": "lib/",
+    "module": "CommonJS",
+    "declaration": true,
+    "sourceMap": true
+    },
+  "files": ["index.ts"]
```
Comment (hurryabit) [inline language-support/ts/daml-react/tsconfig.json]: This seems wrong. There are way more TypeScript files in here.

File: yarn.lock [hunk 5]
```diff
@@ -4014,10 +4014,10 @@ typedarray@^0.0.6:
...
   integrity sha1-hnrHTjhkGHsdPUfZlqeOxciDB3c=
 
-typescript@3.1.6:
-  version "3.1.6"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.1.6.tgz#b6543a83cfc8c2befb3f4c8fba6896f5b0c9be68"
-  integrity sha512-tDMYfVtvpb96msS1lDX9MEdHrW4yOuZ4Kdc4Him9oU796XldPYF/t2+uKoX0BBa0hXXwDlqYQbXY5Rzjzc5hBA==
+typescript@3.7.4:
+  version "3.7.4"
+  resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.7.4.tgz#1743a5ec5fef6a1fa9f3e4708e33c81c73876c19"
+  integrity sha512-A25xv5XCtarLwXpcDNZzCGvW2D1S3/bACratYBx2sax8PefsFhlYmkQicKHvpYflFS8if4zne5zT5kpJ7pzuvw==
```
Comment (hurryabit) [inline yarn.lock]: Why was this file changed?

File: language-support/ts/daml-react/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+    "esModuleInterop": true,
+    "outDir": "lib/",
+    "module": "CommonJS",
+    "declaration": true,
+    "sourceMap": true
+    }
```
Comment (hurryabit) [inline language-support/ts/daml-react/tsconfig.json]: We should have a `"files"` or `"include"` entry here.
```suggestion
  }
```

File: language-support/ts/daml-react/tsconfig.json (hunk 4 above)
Comment (robin-da) [reply language-support/ts/daml-react/tsconfig.json]: this has already been fixed, thanks.

File: yarn.lock (hunk 5 above)
Comment (robin-da) [reply yarn.lock]: for 'conditional chaining' which we are using in daml-react we need a newer typescript compiler which apparently we get from to npm dependencies. I updated it to 3.7.4.

Comment (hurryabit) [reply yarn.lock]: I know. But your change should not affect this file but rather the one in `language-support/ts/packages/yarn.lock`.
